src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/DefaultRazorProjectChangePublisherTest.cs

```diff
@@ -51,8 +52,100 @@ public class DefaultRazorProjectChangePublisherTest : LanguageServerTestBase
     {
         private readonly RazorLogger RazorLogger = Mock.Of<RazorLogger>();
 
+        public DefaultRazorProjectChangePublisherTest()
+        {
+            ProjectSnapshotManager = CreateProjectSnapshotManager(allowNotifyListeners: true);
+        }
+
+        private ProjectSnapshotManagerBase ProjectSnapshotManager { get; }
+
         private ProjectConfigurationFilePathStore ProjectConfigurationFilePathStore { get; } = new DefaultProjectConfigurationFilePathStore();
 
+        [Fact]
+        public void ProjectManager_Changed_NotActive_Noops()
+        {
+            // Arrange
+            var attemptedToSerialize = false;
+            var hostProject = new HostProject("/path/to/project.csproj", RazorConfiguration.Default, rootNamespace: "TestRootNamespace");
+            var hostDocument = new HostDocument("/path/to/file.razor", "file.razor");
+            ProjectSnapshotManager.ProjectAdded(hostProject);
+            var publisher = new TestDefaultRazorProjectChangePublisher(
+                ProjectConfigurationFilePathStore,
+                RazorLogger,
+                onSerializeToFile: (snapshot, configurationFilePath) => attemptedToSerialize = true)
+            {
+                EnqueueDelay = 10,
+            };
+            publisher.Initialize(ProjectSnapshotManager);
+
+            // Act
+            ProjectSnapshotManager.DocumentAdded(hostProject, hostDocument, new EmptyTextLoader(hostDocument.FilePath));
+
+            // Assert
+            Assert.Empty(publisher._deferredPublishTasks);
+            Assert.False(attemptedToSerialize);
+        }
+
+        [Fact]
+        public void ProjectManager_Changed_DocumentOpened_UninitializedProject_NotActive_Noops()
+        {
+            // Arrange
+            var attemptedToSerialize = false;
+            var hostProject = new HostProject("/path/to/project.csproj", RazorConfiguration.Default, rootNamespace: "TestRootNamespace");
+            var hostDocument = new HostDocument("/path/to/file.razor", "file.razor");
+            ProjectSnapshotManager.ProjectAdded(hostProject);
+            ProjectSnapshotManager.DocumentAdded(hostProject, hostDocument, new EmptyTextLoader(hostDocument.FilePath));
+            var publisher = new TestDefaultRazorProjectChangePublisher(
+                ProjectConfigurationFilePathStore,
+                RazorLogger,
+                onSerializeToFile: (snapshot, configurationFilePath) => attemptedToSerialize = true)
```

Comment:
Nit: I like to use the `_` discard when I know I won't be using the parameters of callbacks like this.

Reply:
I think that's valuable in product code; however, in test code i'm less particular about it. purely to avoid having to rebuild again i'm going toleave as is.